### PR TITLE
[OE] Refactor OE-PDP Connection

### LINKED ISSUE #395: [OE] Update `pdpclient` to work with current PDP implementation
### Current Limitation

Currently the `PdpResponse`  in `pdpclient` is wrong format, not matching the current PDP implementation

### Suggested Improvement

Thanikan to add more context

- [ ] Add traceId
- [ ] Refactor `pdpclient` per Go best practices

### Version

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by sthanikan2000**
## Summary
Refactored the OE-PDP connection to Support the Exact DTO for making policy decision request.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactoring (no functional changes)
- [x] Performance improvement
- [ ] Other (please describe):

## Changes Made
- Add policy constants and DTOs for policy decision requests and responses
- Refactor PDP handling in FederateQuery for improved error logging and consent validation
- Refactor PDP response handling in tests to improve clarity and consistency

## Testing
- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist
- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues
Closes #395

### CODE REVIEW

File: exchange/orchestration-engine/federator/federator.go [hunk 1]
```diff
@@ -297,37 +297,54 @@ func (f *Federator) FederateQuery(ctx context.Context, request graphql.Request,
...
 		}
 
 		pdpRequest.RequiredFields = requiredFields
 
 		pdpResponse, err = pdpClient.MakePdpRequest(pdpRequest)
 		if err != nil {
-			logger.Log.Info("PDP request failed", "error", err)
-			return createErrorResponseWithCode("PDP request failed", errors.CodePDPError)
+			logger.Log.Error("PDP request failed", "error", err)
+			return createErrorResponseWithCode("Authorization check failed: "+err.Error(), errors.CodePDPError)
```
Comment (ginaxu1) [inline exchange/orchestration-engine/federator/federator.go]: nit: use `fmt.Errorf` for better error wrapping
```go
return createErrorResponseWithCode(
    fmt.Errorf("authorization check failed: %w", err).Error(),
    errors.CodePDPError,
)
```
